packages/core/content-releases/server/src/utils/index.ts

```diff
@@ -44,3 +44,27 @@ export const getEntryValidStatus = async (
     return false;
   }
 };
+
+/**
+ * Temporal function to get the entryId from documentId with locale
+ * This is needed because some services still need the entryId to work
+ */
+export const getEntryId = async (
+  {
+    contentTypeUid,
+    documentId,
+    locale,
+  }: { contentTypeUid: UID.ContentType; documentId?: string; locale?: string },
+  { strapi }: { strapi: Core.Strapi }
+) => {
+  if (documentId) {
+    const document = await strapi.documents(contentTypeUid).findOne({ documentId, locale });
+
+    return document?.id;
+  }
+
+  // documentId is not defined in singleTypes, so we want to get the first document
+  const documents = await strapi.documents(contentTypeUid).findMany({ locale });
+
+  return documents[0].id;
```

Comment:
Use the `.findFirst` instead, this might load 1M documents if they exist in the database.